### PR TITLE
Release 0.81 version

### LINKED ISSUE #3744: [ANNOUCEMENT] 0.81 release planned on November 1, 2018
We will release version 0.81 on October 15, 2018.

Blocking issues / PRs
- [x] #3523: colsample_bytree seemingly not random
- [x] #3543: Multi-threading not working properly with version 0.71.2
- [x] #3579: hist + learning_rate decay memory usage
- [x] #3643: Lanuch function for unifying CPU and GPU code
- [x] #3738: Multi-GPU support in GPUPredictor
- [x] #3746: Binary wheels are broken on CPU-only containers
- [x] #3750: For training data with group, empty RDD partition threw exception
- [x] #3758:  fix issue when spark job execution thread cannot return before we execute first()
- [x] #3808: fix the early stopping feature in XGBoost4J
- [x] #3809: benchmark/benchmark_tree.py failed
- [x] #3816: Upgrade Scala to 2.11.12 to address security vulnerability
- [x] #3840: `tree_method=approx` invokes the wrong tree updater class

### PR DESCRIPTION
**Pull request by hcho3**
Closes #3744.

### CODE REVIEW

**Comment by CodingCat** [comment]
shall we cut branch_0.81?

**Comment by CodingCat** [comment]
nvm...some browser problem

**Comment by hcho3** [comment]
@CodingCat Here it is: https://github.com/dmlc/xgboost/commits/release_0.81

**Comment by CodingCat** [comment]
sure, I have put jvm jars to maven central...the web page should come in days

**Comment by hcho3** [comment]
@CodingCat Awesome.

BTW, awesome job on XGBoost4J-Spark. As I drafted the release note, I noticed a lot of efforts put into this area. If you're willing, I'd love to visit your team and hear more about your work.

**Comment by trivialfis** [comment]
Sorry that it seems I'm one step too late to look into this. :(
The added `Transform` class brought some objectives functions into GPU like `softmax` and `hinge`.

At the same time, it also deprecated the old GPU specifier like "gpu:reg:linear".
https://github.com/dmlc/xgboost/blob/91cc14ea70d2c51b0e68369e91182e3423e28b68/src/objective/regression_obj.cu#L150

Later on I will try to draft some simple documents for GPU status.

**Comment by hcho3** [comment]
@trivialfis Feel free to modify `NEWS.md` to incorporate your description of `Transform` class. I didn't have much good idea about what it did, so I just said "de-duplicate CPU/GPU code." Maybe you can create a new heading "New feature: additional objective functions for GPUs" or something like that.

**Comment by hcho3** [comment]
@trivialfis Also, for the next release, I will prepare the release draft a bit ahead of time before the release. Or better yet, I will ask you and other contributors for inputs in describing what each PR does. Please forgive me, still learning and growing into the maintainer role.

**Comment by CodingCat** [comment]
we can minic the behavior of apache projects, before every release, we send a release candidate annoucement to that private list, and leave 2-3 days for voting...the content of voting include whether there is blocking issue, whether the release note is complete, etc.

**Comment by CodingCat** [comment]
http://www.apache.org/dev/release-publishing.html   We just need to be cautious that do not let this process become as a burden (like some apache projs)

**Comment by hcho3** [comment]
@CodingCat Got it. From now on, let us put release candidate first and put it to vote. Any blocking issues will be identified then. As for release cycles, I think two month interval is still good, but we can be flexible as circumstances arise. I probably rushed this one because I felt burdened by my earlier ETA of November 1.

**Comment by CodingCat** [comment]
Yes, I think since we started formal release process since 0.7+, the project is attracting more users as they feel more accountability of us, thanks for your volunteering to handle all release process work @hcho3

**Comment by hcho3** [comment]
@CodingCat No problem. Yes, I think having a regular cycle and ETA is essentially a good thing, as long as it is not too rigid. Next time when blocking issues come up, we should publicly adjust our ETA so that we can deliver high-quality releases in reasonable amount of time.

**Comment by trivialfis** [comment]
@hcho3 That sounds really nice.

**Comment by jimmywan** [comment]
Any idea when wheels will be published?

**Comment by hcho3** [comment]
@jimmywan Currently building a wheel for Windows. Give us a few days

**Comment by hcho3** [comment]
@jimmywan The wheels are up now: https://pypi.org/project/xgboost/#files
